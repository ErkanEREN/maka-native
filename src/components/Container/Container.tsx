import React, { ReactNode } from "react";
import { View } from "react-native";
import styled from "styled-components";

interface Props {
  children: ReactNode;
}

export default function Container({ children }: Props) {
  return <View>{children}</View>;
}
