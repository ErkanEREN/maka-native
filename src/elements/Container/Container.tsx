import React, { ReactNode } from 'react'
import { View } from 'react-native'
import styled from 'styled-components'

const ContainerView = styled(View)<{ padding?: string; bgColor?: string }>`
  flex: 1;
  ${({bgColor}) => bgColor && `background-color: ${bgColor};`};
  ${({padding}) => padding && `padding: ${padding};`};
`

interface Props {
  children: ReactNode
  bgColor?: string
  padding?: string
}

export default function({ children, bgColor, padding }: Props) {
  return (
    <ContainerView padding={padding} bgColor={bgColor}>
      {children}
    </ContainerView>
  )
}
