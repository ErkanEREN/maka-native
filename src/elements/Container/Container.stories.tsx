// src/components/Container/Container.stories.tsx

import React from "react";
import { Text } from "react-native";
import { storiesOf } from "@storybook/react-native";

import Container from "./Container";

storiesOf("Container", module).add("Container", () => (
  <Container bgColor="#f1f1f1" padding="3%">
    <Text>Container</Text>
  </Container>
));
