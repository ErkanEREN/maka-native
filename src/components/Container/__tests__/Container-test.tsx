import React from "react";
import { shallow } from "enzyme";
import Container from "../Container";
import { Text } from "react-native";
import renderer from "react-test-renderer";

describe("Container", () => {
  describe("Rendering", () => {
    it("should match to snapshot", () => {
      const children = <Text>Container Children</Text>;
      const tree = renderer.create(<Container>{children}</Container>).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
