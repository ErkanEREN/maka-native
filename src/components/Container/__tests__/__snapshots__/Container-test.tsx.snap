// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container Rendering should match to snapshot 1`] = `
<div>
  <div>
    Container Children
  </div>
</div>
`;
